test/unit/org/apache/cassandra/index/sai/cql/AllowFilteringTest.java

```diff
@@ -296,6 +296,79 @@ public void testAllowFilteringOnClusteringAndRegularColumns() throws Throwable
         test("SELECT * FROM %s WHERE v1=0 AND v2=0 AND k1=0 AND k2=0 AND (c1, c2, c3, c4) = (0, 0, 0, 0) AND v3=0", true);
     }
 
+    @Test
+    public void testAllowFilteringTextWithINClause () throws Throwable
+    {
+        createTable("CREATE TABLE %S (K1 TEXT, K2 TEXT, K3 TEXT, PRIMARY KEY(K1))");
+        createIndex(format("CREATE CUSTOM INDEX ON %%s(K2) USING '%s'", StorageAttachedIndex.class.getName()));
+
+        execute("INSERT INTO %s (K1,K2,K3) VALUES ('s1','s11','s111')");
+        execute("INSERT INTO %s (K1,K2,K3) VALUES ('s2','s11','s11')");
+        execute("INSERT INTO %s (K1,K2,K3) VALUES ('s3','s22','s111')");
+        execute("INSERT INTO %s (K1,K2,K3) VALUES ('s4','s22','s111')");
+        execute("INSERT INTO %s (K1,K2,K3) VALUES ('s5','s31','s111')");
+
+        assertRowCount(execute("SELECT * FROM %s WHERE K2='s11' AND K3 IN ('s11','s111') ALLOW FILTERING"),2);
+        assertRowCount(execute("SELECT * FROM %s WHERE K2='s22' AND K3 IN ('s111','s111') ALLOW FILTERING"), 2);
+
+    }
+
+    @Test
+    public void testAllowFilteringIntWithINClause () throws Throwable
+    {
+        createTable("CREATE TABLE %S (K1 text, K2 int, K3 int, PRIMARY KEY(K1))");
```

Comment:
nit: Not super important, but in all these new tests, let's settle on a consistent lower-casing of the column names? (ex. `k1`, `k2`, `k3`, etc.)